### PR TITLE
Refactored main game screen

### PR DESCRIPTION
**Pull request by RjCor**
Implemented new class system with the main game.
Refactored main game screen code.
Added hints button.
Many, many vfx changes.

### CODE REVIEW

File: MVC/View_GUI/gui_main_game.py [hunk 1]
```diff
@@ -1,91 +1,82 @@
...
 from model_puzzle import *
 
-class Game:
-    def __init__(self, puzzle, puzzle_stats):
-            
-        self.puzzle = puzzle
-        self.puzzle_stats = puzzle_stats
-        print(f'gui_main_game.py - def__init__(): puzzle.pangram is: {puzzle.pangram}')
-        print(f'gui_main_game.py - def__init__(): puzzle.required_letter is: {puzzle.required_letter}')
+# from model_shuffleLetters import *
```
Comment (EthanWright24) [inline MVC/View_GUI/gui_main_game.py:4]: can you remove these lines (4-7)

File: MVC/View_GUI/gui_main_game.py [hunk 2]
```diff
@@ -234,270 +202,260 @@ def draw_hexagon(self):
...
-                    break
-            
-            pygame.draw.rect(self.game_window, self.WHITE, file_input_rectangle)
-            # Render the text as a Surface and blit it onto the text display
-            text_surface = font.render(input_text, True, (0, 0, 0))
-            self.game_window.blit(text_surface, (file_input_x, file_input_y))
-            pygame.display.update()
-        
+        pass
+        # semi_transparent_color = (0, 0, 0, 160)
```
Comment (EthanWright24) [inline MVC/View_GUI/gui_main_game.py:390]: Should this be left in? 390-440

Comment (RjCor) [reply MVC/View_GUI/gui_main_game.py:390]: Yep, implementing save game currently.

File: MVC/View_GUI/gui_main_game.py (hunk 1 above)
Comment (RjCor) [reply MVC/View_GUI/gui_main_game.py]: They have been removed

**Review by EthanWright24 — APPROVED** [review]

**Pull request merged**